internal/bucket/access.go

```diff
@@ -47,3 +47,19 @@ func (b *bucketImpl) GetBuckets() []*BucketContent {
 
 	return content
 }
+
+func (b *bucketImpl) ResolveContentBySchema(did string) ([]*BucketContent, error) {
+	var matchedContent []*BucketContent = make([]*BucketContent, 0)
+	for _, c := range b.contentCache {
+		if c.ContentType == bt.ResourceIdentifier_CID {
+			bi, ok := c.Item.(*bt.BucketItem)
+			if ok {
```

Comment:
nit: You can reduce nesting by doing `if ok && bi.SchemaDefinition.Did == did {`

You could even combine the above line (if you want) to 
```
if bi, ok := c.Item.(*bt.BucketItem); ok && bi.SchemaDefinition.Did == did {
}
```

Reply:
i went for the above to improve readability of the conditions but what you are proposing isnt that bad. ill update